test/run_test.sh

```diff
@@ -0,0 +1,288 @@
+#!/bin/bash
+TS_TMPDIR=
+TS_QUEUE_RESULT=
+TS_EXIT_CODE=0
+TS_LOG_DEBUG=false
+TS_TEST_DIR=
+TS_DISPLAY_TEST_LOG=true
+TS_DISPLAY_TEST_RESULT=true
+TS_SYNC_TEST=false
+TS_TEST_TO_RUN=
+
+help() {
+    cat <<EOF
+$0 [-d][-q][-Q][-s][-o <test_filename>] <test_dir>
+
+Help:
+-d debug log
+-h display this message
+-q quiet tests log
+-Q quiet tests result
+-s run test in a sync way
+-t specify one test filename to run. This file should be in <test_dir>
+EOF
+    exit 0
+}
+
+# Check if all needed environment variable are set
+check_env() {
+    while getopts ":dhqQst:" option
+    do
+        case $option in
+            d)
+                TS_LOG_DEBUG=true
+            ;;
+            h)
+                help
+            ;;
+            q)
+                TS_DISPLAY_TEST_LOG=false
+            ;;
+            Q)
+                TS_DISPLAY_TEST_RESULT=false
+            ;;
+            s)
+                TS_SYNC_TEST=true
+            ;;
+            t)
+                TS_TEST_TO_RUN=$OPTARG
+            ;;
+            :)
+                echo "Arg required for $OPTARG"
+                exit 1
+            ;;
+            \?)
+                echo "Invalid option: $OPTARG"
+                echo
+                help
+            ;;
+        esac
+    done
+    
+    # First args after option should be the TS_TEST_DIR
+    shift $((OPTIND-1))
+    TS_TEST_DIR=$1
+    if [[ -z $TS_TEST_DIR ]];then
+        echo "Test directory not provide"
+        help
+    fi
+    if [[ ! -d $TS_TEST_DIR ]];then
+        log_error "Test directory $TS_TEST_DIR not found"
+    fi
+    
+    # Check test files are OK
+    get_test_files > /dev/null
+}
+
+# Log
+log() {
+    echo "INFO: $@"
+}
+
+log_error() {
+    echo "ERROR: $@" >&2
+    exit 1
```

Comment:
Maybe move the `exit` out of `log_error`, to a function with a more explicit name which will call `log_error` before exiting.